repos/exporter/lib/atcoCheckerHandler.ts

```diff
@@ -0,0 +1,286 @@
+import { Handler } from 'aws-lambda';
+import { S3 } from 'aws-sdk';
+import { WithIds, ReturnTicket, SingleTicket, PointToPointPeriodTicket } from '../shared/matchingJsonTypes';
+import {
+    saveIdsOfServicesRequiringAttentionInTheDb,
+    getDirectionAndStopsByLineIdAndNoc,
+    getPointToPointProducts,
+} from './database';
+import { ExportLambdaBody } from '../shared/integrationTypes';
+import { DirectionAndStops } from '../shared/dbTypes';
+
+const s3: S3 = new S3(
+    process.env.NODE_ENV === 'development'
+        ? {
+              s3ForcePathStyle: true,
+              accessKeyId: 'S3RVER',
+              secretAccessKey: 'S3RVER',
+              endpoint: 'http://localhost:4572',
+          }
+        : {
+              region: 'eu-west-2',
+          },
+);
+
+const PRODUCTS_BUCKET = process.env.PRODUCTS_BUCKET;
+
+export const handler: Handler<ExportLambdaBody> = async () => {
+    console.log('triggered atco checker lambda...');
+
+    if (!PRODUCTS_BUCKET) {
+        throw new Error('Need to set PRODUCTS_BUCKET env variable');
+    }
+
+    const pointToPointProducts: {
+        id: number;
+        lineId: string;
+        matchingJsonLink: string;
+        startDate: string;
+        endDate: string;
+    }[] = await getPointToPointProducts();
+
+    for (let i = 0; i < pointToPointProducts.length; i++) {
+        const pointToPointProduct = pointToPointProducts[i];
+        const productId: number = pointToPointProduct.id;
+        const path = pointToPointProduct.matchingJsonLink;
+
+        const object = await s3.getObject({ Key: path, Bucket: PRODUCTS_BUCKET }).promise();
+
+        if (!object.Body) {
+            throw new Error(`body was not present [${path}]`);
+        }
+
+        const pointToPointTicket = JSON.parse(object.Body.toString('utf-8')) as
+            | WithIds<SingleTicket>
+            | WithIds<ReturnTicket>
+            | WithIds<PointToPointPeriodTicket>;
+
+        // check to see if single ticket
+        if ('fareZones' in pointToPointTicket) {
+            const idsOfServicesRequiringAttention: string[] = await getIdsOfServicesRequiringAttentionForSingles(
+                pointToPointTicket,
+            );
+
+            await saveIdsOfServicesRequiringAttentionInTheDb(productId, idsOfServicesRequiringAttention);
+        } else if ('inboundFareZones' in pointToPointTicket && 'outboundFareZones' in pointToPointTicket) {
+            // we have a return ticket
+            const idsOfServicesRequiringAttention: string[] = await getIdsOfServicesRequiringAttentionForReturns(
+                pointToPointTicket,
+            );
+
+            await saveIdsOfServicesRequiringAttentionInTheDb(productId, idsOfServicesRequiringAttention);
+        }
+    }
+};
+
+/**
+ * For single tickets; gets a list of ids for the serices that require
+ * attention because of a change on the stops relating to the service.
+ *
+ * @param {pointToPointTicket} the matching json for single ticket.
```

Comment:
Might be worth changing this variable from pointToPointTicket to singleTicket - as we dont accept any other type.